### DISCUSSION TITLE
Next js App Directory Locale and BasePath Change

### DISCUSSION THREAD

**Discussion by AwaisTariq18** [vercel/next.js#45513] [Help] [answered]
I'm following a guide provided by next js to implement Internationalization. how can i update by basePath e.g. if i have language switcher and my default locale is en and my path will become localhost:3000/en and by switching it to fr i want my basePath to update to localhost:3000/fr like that so that whenever i open a new page lets say about-us my basePath should start from localhost:3000/fr instead of localhost:3000/en which is a default one.

**Comment by guitheengineer** [comment, marked as answer]
You can set the language chosen on local storage so if the user access the `/` page, you redirect them to that stored locale path

**Reply by AwaisTariq18** [reply]
Thanks! I managed to do this with NEXT_LOCALE cookie
